internal/getproviders/package_location_local_archive.go

```
@@ -67,6 +81,8 @@ func (p PackageLocalArchive) InstallProviderPackage(_ context.Context, meta Pack
 	//nolint:mnd // magic number predates us using this linter
 	err := unzip.Decompress(targetDir, filename, true, 0000)
 	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
```

I was thinking if it would be a good idea to include a function on the internal/tracing library for calling these error recordings, like:

tracing.SetError(err)

```
SetError(span trace.Span, err error) {
  span.SetStatus(codes.Error, err.Error());
  span.RecordError(err)
}
```

Just so we could remove some duplication of the code. Let me know what you think about it.

---

I was thikning about this too, but then I also realised that there are some situations where we want to report the error from a diag or something else, so i decided to just go with this method. but I'm 50/50 on this.

---

I agree with Dio here. If we want to do the diags thing, we could go with a similar implementation of the `Diagnostics.Append` where it checks the type and handles that internally.

---

I've added this method in the utils file here: https://github.com/opentofu/opentofu/pull/2665/files#diff-338d0067ba5fd1e76a34f61066eecaed4b9688f08e3869682fa04c0c05305c3cR40-R70

Let me know what you guys think @yottta / @diofeher